cmd/config_merger/main.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"time"
+
+	"github.com/GoogleCloudPlatform/testgrid/pkg/merger"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	listPath     string
+	creds        string
+	confirm      bool
+	wait         time.Duration
+	skipValidate bool
+}
+
+func (o *options) validate() error {
+	if o.listPath == "" {
+		return errors.New("--config-list of configurations to merge required")
+	}
+	if !o.confirm {
+		logrus.Info("--confirm=false (DRY-RUN): will not write to gcs")
+	}
+	if o.skipValidate {
+		logrus.Info("--allow-invalid-configs: result may not validate either")
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	var o options
+	flag.StringVar(&o.listPath, "config-list", "", "List of configurations to merge")
```

Comment:
Kind of surprised this refers to a typed yaml document, rather than a comma delimited list of config paths and/or a repeatable flag.

Reply:
I really wanted to include at least a "contact" field for entries, since a target use case for this is multiple teams merging their configurations onto one TG instance.

If one team's configuration is deleted or corrupted, we need a way to surface that to them as soon as possible without breaking every other tenant. Ideally, their presubmits would prevent this, but "please give us a way to alert you" seems like a good precaution. 

I also expect ParseAndCheck to be sufficient as a presubmit to check this yaml config file for obvious errors, but it's not obvious that the data at a valid path is invalid until you actually check.